src/Controls/src/Core/Compatibility/Handlers/ListView/iOS/ContextActionCell.cs

```diff
@@ -340,15 +340,18 @@ void ActivateMore()
 			if (controller == null)
 				throw new InvalidOperationException("No UIViewController found to present.");
 
-			if (UIDevice.CurrentDevice.UserInterfaceIdiom == UIUserInterfaceIdiom.Phone)
+			if (UIDevice.CurrentDevice.UserInterfaceIdiom == UIUserInterfaceIdiom.Phone || (UIDevice.CurrentDevice.UserInterfaceIdiom == UIUserInterfaceIdiom.Pad && actionSheet.PopoverPresentationController == null))
 			{
 				var cancel = UIAlertAction.Create(StringResources.Cancel, UIAlertActionStyle.Cancel, null);
 				actionSheet.AddAction(cancel);
 			}
 			else
 			{
-				actionSheet.PopoverPresentationController.SourceView = _tableView;
-				actionSheet.PopoverPresentationController.SourceRect = sourceRect;
+				if (actionSheet.PopoverPresentationController != null)
+				{
+					actionSheet.PopoverPresentationController.SourceView = _tableView;
+					actionSheet.PopoverPresentationController.SourceRect = sourceRect;
+				}
 			}
```

Comment:
Will it cause issues to simplify this to be:

```cs
if (Phone || actionSheet.PopoverPresentationController is null)
{
    var cancel = UIAlertAction.Create(StringResources.Cancel, UIAlertActionStyle.Cancel, null);
    actionSheet.AddAction(cancel);
}
else // actionSheet.PopoverPresentationController cannot be null as it would have made it to the block above
{
    actionSheet.PopoverPresentationController.SourceView = _tableView;
    actionSheet.PopoverPresentationController.SourceRect = sourceRect;
}
```

This will be the same in this file and src/Compatibility/Core/src/iOS/ContextActionCell.cs